foreshadow/concrete/internals/cleaners/base.py

```diff
@@ -213,13 +201,15 @@ def transform(self, X, y=None):
                 )
             columns = self.output_columns
             if columns is None:
-                columns = [
-                    X.columns[0] + str(c) for c in range(len(out.iloc[0]))
-                ]
                 # by default, pandas would have given a unique integer to
                 # each column, instead, we keep the previous column name and
                 # add that integer.
-            X = pd.DataFrame([*out.values], columns=columns)
+                columns = [
+                    X.columns[0] + str(c) for c in range(len(out.iloc[0]))
+                ]
+            # We need to set the index. Otherwise, the new data frame might
+            # misalign with other columns.
+            X = pd.DataFrame([*out.values], index=out.index, columns=columns)
```

Comment:
Check whether this can be simplified to
```python
X = out.rename(columns=columns)
```

Reply:
It turns out that `out` is a `pd.Series`, not a `pd.Dataframe`.  Therefore we need to stick to the original code.